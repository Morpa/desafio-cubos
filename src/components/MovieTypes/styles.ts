import { shade } from 'polished'
import styled, { css } from 'styled-components'

export const Wrapper = styled.span`
  ${({ theme }) => css`
    padding: 0.2rem 0.9rem;
    text-align: center;
    background: ${theme.colors.white};
    border: 2px solid ${theme.colors.primary};
    border-radius: 2rem;
    color: ${theme.colors.primary};
    transition: background ${theme.transition.default};

    &:hover {
      background: ${shade(0.1, theme.colors.white)};
    }
  `}
`
