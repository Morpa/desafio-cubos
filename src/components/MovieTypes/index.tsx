import * as S from './styles'

export type MovieTypesProps = {
  name: string
}

const MovieTypes = ({ name }: MovieTypesProps) => <S.Wrapper>{name}</S.Wrapper>

export default MovieTypes
