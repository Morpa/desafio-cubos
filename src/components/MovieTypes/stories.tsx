import { Story, Meta } from '@storybook/react/types-6-0'
import MovieTypes, { MovieTypesProps } from '.'

export default {
  title: 'MovieTypes',
  component: MovieTypes,
  args: {
    name: 'Terror'
  }
} as Meta

export const Default: Story<MovieTypesProps> = (args) => (
  <MovieTypes {...args} />
)
