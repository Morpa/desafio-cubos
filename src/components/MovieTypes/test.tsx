import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import MovieTypes from '.'

describe('<MovieTypes />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<MovieTypes name={'Suspense'} />)

    expect(screen.getByText(/suspense/i)).toBeInTheDocument()

    expect(screen.getByText(/suspense/i)).toHaveStyleRule(
      'background',
      '#e5e5e5',
      {
        modifier: ':hover'
      }
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
