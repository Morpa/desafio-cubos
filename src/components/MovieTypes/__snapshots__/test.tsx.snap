// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieTypes /> should render correctly 1`] = `
.c0 {
  padding: 0.2rem 0.9rem;
  text-align: center;
  background: #FFF;
  border: 2px solid #116193;
  border-radius: 2rem;
  color: #116193;
  -webkit-transition: background 0.3s ease-in-out;
  transition: background 0.3s ease-in-out;
}

.c0:hover {
  background: #e5e5e5;
}

<span
  class="c0"
>
  Suspense
</span>
`;
